compiler/codegen/codegen.cpp

```diff
@@ -2147,6 +2192,17 @@ void codegen(void) {
                 executableFilename);
     }
     *lastDot = '\0';
+
+  }
+
+  if (fLibraryCompile && libmodeHeadername[0] == '\0') {
+    if (strncmp(executableFilename, "lib", 3) == 0) {
+      strncpy(libmodeHeadername, executableFilename + 3,
+              sizeof(executableFilename) - 3);
```

Comment:
What will this do if the user's compiling `lib.chpl`?

Reply:
If they're relying on the default -o settings, probably generate a file named just `.h`.  But in that case you'd get a library named `lib.a` and I think you can't specify that via -l?

Reply:
At least, I don't seem to be able to do so

Reply:
I'm really confused about the purpose of this code. I'd think it'd be better to not try to strip off `lib` from `libmymodule.chpl`.

Reply:
Cython expects the header file name and the library file name to have a relationship of "moduleName.h" to "libmoduleName.a/.so".  Our generated library file name does not include the lib prefix by default, so generally what I've had to do is explicitly specify it using -o libmoduleName.  I chose to base the name for the header file off the executableFilename so I didn't have to separately compute the name and risk code drift should we decide to change the generated name for the .a, but that means that I have to strip the lib portion back off for the header name.

I could change our generated name in library mode includes the lib prefix (since that seems like what you would expect in that case anyways) and avoid updating the header name when that happens.  Do you think that would be appropriate to do in this PR?

Reply:
> I could change our generated name in library mode includes the lib prefix (since that seems like what you would expect in that case anyways) and avoid updating the header name when that happens. Do you think that would be appropriate to do in this PR?

I think you're talking about the default generated name here, right? Or all the time? Since header mode makes another file, maybe we want to be able to specify the name for it separately as its own flag?

I don't know how much testing impact that does - if it changes a lot of tests separate PR is better.

Reply:
> I think you're talking about the default generated name here, right? Or all the time?

Yeah, the default generated name which can get overridden by `-o`.  I would want to only add a "lib" prefix to it in `--library` compilation.

> Since header mode makes another file, maybe we want to be able to specify the name for it separately as its own flag?

This is certainly not a good reason to avoid it, but won't that mean adding special casing to start_test/sub_test for library compiles so that we don't override the default?  We still explicitly pass `-o` there.

Changing the behavior of `-o` for library mode versus normal compilation doesn't give me pause, since you would only be able to use at most one of the two flags in any given compile (there's no world where you would want to be able to set both in a single compilation, they impact roughly the same thing, I don't particularly see the point of having a separate flag for it).  I would be curious if others felt that was the right choice.

> I don't know how much testing impact that does - if it changes a lot of tests separate PR is better.

I believe every test that it would impact was already impacted by this change (due to preventing git from tracking the header files I am generating, and cleaning up the generated files)